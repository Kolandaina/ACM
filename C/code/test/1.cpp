#include<bits/stdc++.h>
#include<functional>
#define int long long
#define endl "\n"
const int N = 2e5 + 10;
const int mod = 1e9 + 7;
const int INF = INT64_MAX;
#define fs(x) std::fixed << std::setprecision(x)
inline void solve() {
    int n, x;
    std::cin >> n >> x;
    std::vector<std::vector<int>> g(n + 1);
    std::vector<int> in(n + 1);
    for (int i = 1; i < n; ++i) {
        int u, v;
        std::cin >> u >> v;
        g[u].push_back(v);
        g[v].push_back(u);
        in[u]++, in[v]++;
    }
    if (n == 1)std::cout << "Ayush" << endl;
    else {
        if (in[x] == 1)std::cout << "Ayush" << endl;
        else if (n % 2)std::cout << "Ashish" << endl;
        else std::cout << "Ayush" << endl;
    }
}
signed main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    int t = 1;
    std:: cin >> t;
    while (t--) solve();
    return 0;
}