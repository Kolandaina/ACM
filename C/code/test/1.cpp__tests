[
	{
		"test": "4 5\n",
		"uncorrect_answers":
		[
			"1/4"
		]
	},
	{
		"test": "4 8\n"
	}
]
