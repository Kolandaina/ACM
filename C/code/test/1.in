7 7 10
1 6
6 7
7 3
3 4
4 2
2 5
5 7