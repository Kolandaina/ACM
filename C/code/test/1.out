2 3
.#.
.B.
2 3
#G#
BB.
2 3
G.#
B#.
2 3
##.
GB.
2 3
#G.
.B.
2 3
...
B#.
2 3
#GG
...
2 3
#.B
GB.
2 3
B.G
.G.
2 3
BB#
##.
2 3
#G.
G#.
2 3
B.B
##.
2 3
BBB
GG.
2 3
BBG
##.
2 3
BB#
GB.
2 3
GG.
G#.
2 3
#.#
G#.
2 3
G..
...
2 3
GG.
.#.
2 3
B.#
B..
2 3
#GB
#G.
2 3
##G
#G.
2 3
G#B
#..
2 3
.B.
B..
2 3
.#.
#B.
2 3
G#.
B..
2 3
G#.
#B.
2 3
G..
#B.
2 3
.B#
GG.
2 3
G##
#B.
2 3
#G#
G#.
2 3
.B#
#..
2 3
GBG
#G.